sourcegraph/learn
# Add guide content type & Header and HeaderCta components

### What should this PR do?
Resolves [this Asana ticket](https://app.asana.com/0/1200198213175654/1200636734706520/f) by adding a new type `guides` and the `code-search` guide.

Also adds two new components (which I would anticipate becoming `atoms` following that refactor): `Header` and `HeaderCta`.

Refactors our record loading `util` functions to accommodate another record type; also refactors the author data flow to simplify how we handle that data for now.

### Why are we making this change?
- We will be working with multiple content types in the future; this refactor is the beginning of achieving greater flexibility in the app in order to do that.
- We would like to be able to feature the code search guide from the home page once it is ready, so the `HeaderCta` component sets us up for that.

### What are the acceptance criteria? 
- All `post` records display correctly following refactor.
- All authors (of both `posts` and `guides` display on the authors index page.

### How should this PR be tested?
- Check out the branch, and test all existing records to ensure above.
- Check authors index page.

## Pull request process

**Reviewers**:

1. Test functionality using the criteria above.
2. Offer tips for efficiency, feedback on best practices, and possible alternative approaches and things that may not have been considered.
3. For shorter, "quick" PRs, use your best judgement on #​2.
4. Use a collaborative approach and provide resources and/or context where appropriate.
5. Provide screenshots/grabs where appropriate to show findings during review.

**Reviewees**:

1. Prefer incremental and appropriately-scoped changes.
2. Leave a comment on things you want explicit feedback on.
3. Respond clearly to comments and questions.
